### PR TITLE
disable max stream duration

### PR DESCRIPTION
**Pull request by ramaraochavali**
This PR https://github.com/istio/istio/pull/28089 incorrectly sets the max stream duration to the timeout value provided. The side affect of that is if the default mesh configuration configures the timeout to 20s, we set the max stream duration also to 20s and if clients set this x-envoy-upstream-rq-timeout-ms header to specify explicit timeout value larger than 20s, stream is terminated at 20s not honouring the envoy header. 

We ran in to this regression with latest upgrade. This PR disables stream duration so that the timeout is driven by route timeout/gRPC timeout and can be overridden with envoy header or grpc-timeout header.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by adiprerepa — APPROVED** [review]
LGTM

File: releasenotes/notes/max-stream-duration.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: networking
+issue: []
```
Comment (shamsher31) [inline releasenotes/notes/max-stream-duration.yaml:4]: If the issue number is not available, we can add the PR id here so the release notes generator will link PR in the release notes.

```suggestion
issue: 30885
```

Comment (ramaraochavali) [reply releasenotes/notes/max-stream-duration.yaml]: Ok. thank you. Added

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 2]
```diff
@@ -418,7 +418,8 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 			if action.MaxStreamDuration == nil {
 				action.MaxStreamDuration = &route.RouteAction_MaxStreamDuration{}
 			}
-			action.MaxStreamDuration.MaxStreamDuration = d
+			// Disable MaxStream duration to let the request timeout/gRPC timeout drive the timeout for request.
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/route/route.go:421]: MaxStreamDuration is used to reset client connections periodically on running streams https://www.envoyproxy.io/docs/envoy/latest/faq/configuration/timeouts.html?highlight=max_stream_duration#stream-timeouts - Not used for controlling request timeouts. Request timeouts are controlled by request timeout or MaxGrpctimeout (to honour grpc-timeout header capped to this value)

File: releasenotes/notes/30885.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: networking
+issue: [30885]
```
Comment (shamsher31) [inline releasenotes/notes/30885.yaml:4]:
```suggestion
issue: 30885
```

**Review by hzxuzhonghu — APPROVED** [review]
We should backport the fix

File: pilot/pkg/networking/core/v1alpha3/route/route.go (hunk 2 above)
Comment (lambdai) [reply pilot/pkg/networking/core/v1alpha3/route/route.go:421]: I see... we didn't set max stream duration in the past.

**Review by lambdai — APPROVED** [review]
Thank you for  the fix!

**Comment by ramaraochavali** [comment]
@howardjohn Can you PTAL as well?

**Review by shamsher31 — APPROVED** [review]
Thanks.

**Pull request merged**